yarn-project/end-to-end/src/e2e_synching.test.ts

```diff
@@ -0,0 +1,428 @@
+/*
+ * @note  The following test have two modes:
+ *        - Setup where we are constructing blocks and saving for later usage
+ *        - Execution where we will load and execute the blocks, then try a fresh
+ *          sync.
+ *
+ *        The split is made since constructing the number of blocks that we desire
+ *        takes an eternity. Especially for complex transactions will building the
+ *        block take a significant amount of time.
+ *        With the current test configuration setup run for ~30-40 minutes.
+ *
+ *        To run the Setup run with the `AZTEC_GENERATE_TEST_DATA=1` flag. Without
+ *        this flag, we will run in execution.
+ *
+ *        There is functionality to store the `stats` of a sync, but currently we
+ *        will simply be writing it to the log instead.
+ *
+ *        Notice that we are jumping far into the future such that the setup always
+ *        happen at the same time, making it a lot simpler for us to construct
+ *        blocks that we can replay.
+ *
+ * @dev   Since the test is currently not having any limits on time, we don't gain
+ *        much from running it in CI, and it is therefore skipped.
+ *
+ *
+ * blockCount: 10, txCount: 36, complexity: Deployment:      {"numberOfBlocks":16, "syncTime":17.490706521987914, "numberOfTransactions":366}
+ * blockCount: 10, txCount: 36, complexity: PrivateTransfer: {"numberOfBlocks":19, "syncTime":20.846745924949644, "numberOfTransactions":474}
+ * blockCount: 10, txCount: 36, complexity: PublicTransfer:  {"numberOfBlocks":18, "syncTime":21.340179460525512, "numberOfTransactions":438}
+ * blockCount: 10, txCount: 9,  complexity: MaxDiff:         {"numberOfBlocks":17, "syncTime":49.40888188171387,  "numberOfTransactions":105}
```

Comment:
What's this? What's the difference between blockCount and numberOfBlocks?

Reply:
Will extend comments. `blockCount`: how many test blocks I added, `numberOfblocks`: the total number of blocks, e.g., including deploying canonicals and preparing for the tests.